### ISSUE TITLE
NervesHubWebCore.Workers.DeleteFirmware worker fails every 15 minutes

### ISSUE THREAD

**Issue by pojiro** [nerves-hub/nerves_hub_web#763]
**Describe the bug**
[NervesHubWebCore.Workers.DeleteFirmware](https://github.com/nerves-hub/nerves_hub_web/blob/f6fe2a50a06604b7e717439a60ceab8f3e19c7ee/apps/nerves_hub_web_core/lib/nerves_hub_web_core/workers/delete_firmware.ex) worker fails every 15 minutes.

Below is a log in oban_jobs table. This worker is called every 15 minutes by NervesHubWebCore.Worker.

```
The following arguments were given to NervesHubWebCore.Firmwares.Upload.S3.delete_file/1:

    # 1
    %{}

    (nerves_hub_web_core 0.1.0) lib/nerves_hub_web_core/firmwares/upload/s3.ex:39: NervesHubWebCore.Firmwares.Upload.S3.delete_file/1
    (oban 2.6.1) lib/oban/queue/executor.ex:171: Oban.Queue.Executor.perform_inline/1
    (oban 2.6.1) lib/oban/queue/executor.ex:159: Oban.Queue.Executor.perform_inline/1
    (oban 2.6.1) lib/oban/queue/executor.ex:82: Oban.Queue.Executor.call/1
    (elixir 1.11.3) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
    (elixir 1.11.3) lib/task/supervised.ex:35: Task.Supervised.reply/5
    (stdlib 3.13.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

**Expected behavior**

I have no idea. PR #676 added this worker [here](https://github.com/nerves-hub/nerves_hub_web/pull/676/files#diff-881d93991307c22694cf8135e56797ffae78645016d58c9999dc744afb9677ce) and  [used here](https://github.com/nerves-hub/nerves_hub_web/pull/676/files#diff-7fbf6793949151c40842596aa4832348cd25684cf7b824385caa6b4196e69268).
The original intent may be such that it is only executed when it is needed.

**Comment by pojiro** [comment]
Hi @jjcarstens, @brianberlin.

I'd like to resolve this issue. Could you tell me the reason, 
* What is the intent of using NervesHubWebCore.Workers to remove the storage firmware?
* Is there a problem if calling delete_file of uploader(Application.fetch_env!(:nerves_hub_web_core, :firmware_upload)) for directly?

Thank you.

**Comment by brianberlin** [comment]
hey @pojiro 

IIRC the reason I decided to do the delete in an oban job was so it would work nicely in a Multi.merge. 

there's a comment on the pr here: https://github.com/nerves-hub/nerves_hub_web/pull/676/files#r565374996

**Comment by pojiro** [comment]
Thanks for the reply, @brianberlin.

It's intended to avoid the failure of removing the firmware(from s3) by making Oban try repeatedly, isn't it?

I think what we want to do is to delete firmwares associated the account will be deleted, so I think it would be better to pass the job as one-time than as repeatable like NervesHubWebCore.Worker.

@jjcarstens What do you think?
In this case, should make the Worker support one time job? Or should throw the Oban job directly?

**Comment by pojiro** [comment]
@jjcarstens 
Sorry for the late reply and fix, I make PR #775 . Could you check it?
I choose throw Oban job directly, because NervesHubWebCore.Workers seemed to be optimized for cron.
